switchbot/devices/lock.py

```diff
@@ -86,77 +83,97 @@ async def verify_encryption_key(
 
         return lock_info is not None
 
+    @staticmethod
+    async def api_request(
+        session: aiohttp.ClientSession,
+        subdomain: str,
+        path: str,
+        data: dict = None,
+        headers: dict = None,
+    ) -> dict:
+        url = f"https://{subdomain}.{SWITCHBOT_APP_API_BASE_URL}/{path}"
+        async with session.post(url, json=data, headers=headers) as result:
```

Comment:
Didn't see it till now but I just realized the timeout is removed, and the default timeout here is 300s if not passed. That might be a bit long.

Reply:
I agree, I thought since the session is passed from outside it's going to be taken care of there but if that's not the case we should add one here. Here is a PR for that #238